src/SafeMessageHandlers/README.md

```diff
@@ -0,0 +1,17 @@
+# Safe Message Handlers
+
+This sample shows a workflow using `Temporalio.Workflows.Semaphore` to atomically process certain blocks of workflow
+code to prevent data race issues. The sample code demonstrates assigning cluster nodes to jobs atomically.
```

Comment:
For some strict definition of data race, that can't happen in Temporal regardless (what's being prevented here is I think more strictly a logical race). I'd suggest we say "to serialize execution of certain blocks of workflow code" just to be extra clear here.